Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -42,74 +47,39 @@ extension HTTPClient {
-            }
-
-            guard nameAndValue.count == 2 else {
+            guard let (trimmedName, trimmedValue) = keyValuePair.parseKeyValuePair() else {
                 return nil
             }
-
-            self.name = nameAndValue[0]
-            self.value = nameAndValue[1].omittingQuotes()
-
-            guard !self.name.isEmpty else {
+            guard !trimmedName.isEmpty else {
                 return nil
             }
 
+            self.name = String(utf8Slice: trimmedName, in: header)
+            self.value = String(utf8Slice: trimmedValue.trimmingPairedASCIIQuote(), in: header)
             self.path = "/"
             self.domain = defaultDomain
-            self.expires = nil
+            self.expires_timestamp = nil
             self.maxAge = nil
             self.httpOnly = false
             self.secure = false
 
-            for component in components[1...] {
-                switch self.parseComponent(component) {
-                case (nil, nil):
-                    continue
-                case ("path", .some(let value)):
-                    self.path = value
-                case ("domain", .some(let value)):
-                    self.domain = value
-                case ("expires", let value):
-                    guard let value = value else {
-                        continue
-                    }
-
-                    let formatter = DateFormatter()
-                    formatter.locale = Locale(identifier: "en_US")
-                    formatter.timeZone = TimeZone(identifier: "GMT")
-
-                    formatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss z"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                        continue
-                    }
-
-                    formatter.dateFormat = "EEE, dd-MMM-yy HH:mm:ss z"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                        continue
-                    }
-
-                    formatter.dateFormat = "EEE MMM d hh:mm:s yyyy"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                    }
-                case ("max-age", let value):
-                    self.maxAge = value.flatMap(Int.init)
-                case ("secure", nil):
+            for component in components {
+                switch component.parseCookieComponent() {
+                case ("path", .some(let value))?:
+                    self.path = String(utf8Slice: value, in: header)
+                case ("domain", .some(let value))?:
+                    self.domain = String(utf8Slice: value, in: header)
+                case ("expires", .some(let value))?:
+                    self.expires_timestamp = parseCookieTime(value, in: header)
+                case ("max-age", .some(let value))?:
+                    self.maxAge = Int(String(utf8Slice: value, in: header))
```

Comment:
Can we avoid reifying the `String` here? We can construct `Int` from an arbitrary `StringProtocol`, which will save us the conversion cost.